tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/ComponentVerificationTests.cs

```diff
@@ -97,6 +98,69 @@ public void ExportsFromSamePartMustApplyToSameCapabilities(Type type)
             }
         }
 
+        [Theory]
+        [ClassData(typeof(ComposablePartDefinitionTestData))]
+        public void CertainExportsMustNotBeMarkedWithDynamicCapabilities(Type type)
+        {
+            string[] contractsWithFixedCapabilities = new string[] {
+                ExportContractNames.VsTypes.ProjectNodeComExtension,
+                "Microsoft.VisualStudio.ProjectSystem.ConfiguredProject.AutoLoad",
+                "Microsoft.VisualStudio.ProjectSystem.Project.AutoLoad",
+            };
+
+            var definition = ComponentComposition.Instance.FindComposablePartDefinition(type);
+            
+            Assert.NotNull(definition);
+
+            // BUG: https://github.com/dotnet/project-system/issues/5519
+            if (definition!.Type.FullName == "Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.Subscriptions.DependenciesSnapshotProvider")
+                return;
+
+            foreach (KeyValuePair<MemberRef, ExportDefinition> exportDefinitionPair in definition!.ExportDefinitions)
+            {
+                ExportDefinition export = exportDefinitionPair.Value;
+                if (contractsWithFixedCapabilities.Contains(export.ContractName) && 
+                    export.Metadata.TryGetValue(nameof(AppliesToAttribute.AppliesTo), out object metadata) && 
+                    metadata is string appliesTo)
+                {
+                    foreach (string capability in GetDynamicCapabilities())
```

Comment:
Since in theory any string is a valid dynamic capability, wouldn't it be better to ensure that all of the capabilities in `appliesTo` are part of FixedCapabilities?

Reply:
Yep good idea.